Profile Page Editing

gabriel-jt opened the pull request:
- Completed the inline editing feature for the admin page.
- Click on edit to make all relevant fields editable, and saving reflects changes on backend + frontend (hot refresh)
- Used MUI style overrides to make styling reflect on prod builds.
- Deprecated ProfileInfo components (but reused the code) since using 2 separate components made it difficult to keep track of the useState/form values.
- **Will implement/address form input validation in a separate PR/could be good intro task for next sem devs**

## Testing
Manually tested by editing the admin info and checked that changes were saved on firebase console.
Edited email and checked if user token would be refreshed/changed, but this results in a forced logout.

## Screen Recording
https://user-images.githubusercontent.com/28970991/147613736-c947d904-b683-4b67-ac36-1261f7a35c96.mov

https://user-images.githubusercontent.com/28970991/147613920-53ff9cac-6285-4a63-87e3-9c90d430a5a6.mov



CC: @jacobjk01 @hukellyy

jacobjk01 commented:
idk if this is a problem for anyone else, but when I click admins, it's redirecting me to users.

gabriel-jt commented:
> idk if this is a problem for anyone else, but when I click admins, it's redirecting me to users.

There was an issue with the data on firebase console since some admin documents were missing attributes. The data has been fixed and the admins page should load now (already talked to Jacob about this, but replying here so you're also up to date @hukellyy)

danashimessele reviewed (approved):
Gabe you are a legend

hukellyy commented on components/AdminProfileForm/AdminProfileForm.module.css:
```diff
@@ -0,0 +1,104 @@
+
```
some styling issues on these profile blocks (default when not editing):
- make "About" text 18px and black in colour (should be h4 in global styling and noirish)
- make line under "About" light grey (--grey-light) instead of black
- make sure spacing between About and line is 16px
- fonts are off, make sure the header texts is dm-sans (should apply if u use global styling) and the overline texts are in avenir
- make line underneath Cindy's name 2px and --sapphire-light colour
- make sure spacing between cindy's name and line is 23px 
- remove drop shadow on the edit profile button

hukellyy commented on components/AdminProfileForm/AdminProfileForm.module.css:
```diff
@@ -0,0 +1,104 @@
+    width: 30%;
+}
+
+.editable-field {
```
for the editable fields, when you press "edit" is it possible to still make all the fields technically editable at once, but have hover states on them to indicate which field is currently in focused state? my designs in figma model off notion where u can technically edit all the fields at once by clicking into any field, but the states differ in that it highlights grey when u hover over the field, and turns into a white box + drop shadow once you're ac editing. i think this just makes it more intuitive to see what ur editing at a given time, as well as a subtle confirmation once u click out of that focused state.
if this is what u were talking about being too hard to implement tho, dw about it, i might change how the design looks then bc i dont like how it looks w all drop shadows being activated at once.

https://user-images.githubusercontent.com/77181350/147707314-6cd0a420-540a-4a61-96f4-8180696abc70.mov

hukellyy commented on components/AdminProfileForm/AdminProfileForm.module.css:
```diff
@@ -0,0 +1,104 @@
+    width: 30%;
+}
+
+.editable-field {
+    font-family: Avenir;
+    font-size: medium;
+    border: 0px;
+    width: 60%;
+    background-color: white;
+    border-radius: 5px;
+    box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.2);
+    text-align: left;
+    padding: 7px 0 7px 7px;    
+}
```
for the editable fields, when you press "edit" is it possible to still make all the fields technically editable at once, but have hover states on them to indicate which field is currently in focused state? my designs in figma model off notion where u can technically edit all the fields at once by clicking into any field, but the states differ in that it highlights grey when u hover over the field, and turns into a white box + drop shadow once you're ac editing. i think this just makes it more intuitive to see what ur editing at a given time, as well as a subtle confirmation once u click out of that focused state.
if this is what u were talking about being too hard to implement tho, dw about it, i might change how the design looks then bc i dont like how it looks w all drop shadows being activated at once.

https://user-images.githubusercontent.com/77181350/147707314-6cd0a420-540a-4a61-96f4-8180696abc70.mov

hukellyy commented on components/AdminProfileForm/AdminProfileForm.module.css:
```diff
@@ -0,0 +1,104 @@
+    width: 30%;
+}
+
+.editable-field {
+    font-family: Avenir;
+    font-size: medium;
+    border: 0px;
+    width: 60%;
+    background-color: white;
+    border-radius: 5px;
+    box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.2);
+    text-align: left;
+    padding: 7px 0 7px 7px;    
+}
```
can u also make the font colour --grey-dark :)

hukellyy replied:
also make sure the icon and field text is vertically centre aligned  :)

hukellyy commented on components/AdminProfileForm/AdminProfileForm.module.css:
```diff
@@ -0,0 +1,104 @@
+    gap: 5%;
+}
+
+.box {
+    padding: 4%;
+    justify-content: space-between;
+    align-items: center;
+    border: 2px solid var(--sapphire-light);
+    border-radius: 25px;
+    width: 100%;
+    background-color: white;
+}
```
for the fields in this boxes, in editing state can u make sure the boxes don't affect the spacing of the other boxes. like right now all the boxes shift down when u press edit. would it be possible to keep everything static upon pressing edit, only having the boxes appear around editable fields?

gabriel-jt commented:
@jacobjk01 @hukellyy addressed feedback and based revisions on the Figma! Not everything is completely up to date with the buttons, but all of Kelly's feedback on this PR has been addressed

https://user-images.githubusercontent.com/28970991/150721559-fcba74fd-e9bb-424e-82bd-eb630108ef84.mov

hukellyy commented:
first thing: 
can u make the top and bottom padding on these input fields 2-3px more on each side. 

<img width="310" alt="Screen Shot 2022-02-09 at 10 25 31 PM" src="https://user-images.githubusercontent.com/77181350/153349986-db5440cc-df16-468e-a2fc-0d459a3ea90d.png">

hukellyy commented:
this was a change i made during winter break so that's on me lol but can u make there be a reset password button next to edit? this is since we aren't able to see passwords anymore

<img width="814" alt="Screen Shot 2022-02-09 at 10 27 41 PM" src="https://user-images.githubusercontent.com/77181350/153350244-a0ae9fa4-49fc-42a4-963d-a73fb75e84b3.png">

hukellyy commented on pages/profile/index.tsx:
```diff
@@ -6,91 +6,23 @@ import { GetServerSidePropsContext } from "next";
-    const displayinfo = () => {
-      return (
-        <div className={styles.profile}>
-          <div className={styles.namebar}>
-            <h1> Cindo Zhang </h1>
-            <Button variant="contained" className={styles.button}>
-              <Icon type="editpencil" className={styles.editicon}></Icon>
-              Edit
-            </Button>
```
another thing i added after break: put a disabled save button beside edit when there is no info edited yet

<img width="815" alt="Screen Shot 2022-02-09 at 10 30 13 PM" src="https://user-images.githubusercontent.com/77181350/153350521-653420a4-7a68-4e90-b44e-0323e9d2df93.png">

hukellyy commented on components/AdminProfileForm/AdminProfileForm.module.css:
```diff
@@ -0,0 +1,130 @@
+    padding: 0px 0px 0px 10px;    
+}
+
+.info-field {
+    text-align: left;
+    width: 60%;
+    height: 100%;
+    margin-top: 3px;
+    padding-left: 10px;
+    color: var(--grey-dark);
+}
+
```
is it possible to make this hover state? the component (inline editing field) should be in the figma under design system! 

<img width="562" alt="Screen Shot 2022-02-09 at 10 32 40 PM" src="https://user-images.githubusercontent.com/77181350/153350818-745c6da1-4e54-4cdd-bace-8b866ac7aba3.png">

jacobjk01 commented:
looks good gabe i have no comments :)
